src/cpp/rtps/builtin/BuiltinProtocols.cpp

```diff
@@ -143,11 +143,7 @@ bool BuiltinProtocols::initBuiltinProtocols(
     }
 
     // TypeLookupManager
-    if (m_att.typelookup_config.use_client || m_att.typelookup_config.use_server)
-    {
-        tlm_ = new fastdds::dds::builtin::TypeLookupManager(this);
-        tlm_->init_typelookup_service(mp_participantImpl);
-    }
+    tlm_ = new fastdds::dds::builtin::TypeLookupManager(this);
```

Comment:
As the TypeLookupManager is going to be initialized by default, we might consider modifying where is allocated and if it is a pointer or directly a member of the corresponding class. Have you given any thought to this topic?

Reply:
Changed in [523b3e4](https://github.com/eProsima/Fast-DDS/pull/4126/commits/523b3e44f46779bec49ea92cea752085b202fb56)